### DISCUSSION TITLE
AU915 LoraWAN channels not to spec

### DISCUSSION THREAD

**Discussion by randy197** [jgromes/RadioLib#1307] [Q&A] [answered]
G'day all,

I'm using an SX1276 with my Chirpstack gateway running a RAK5146 concentrator.

On startup with the AU915 region selected the node often selects the wrong frequency bands for up/down links. I have set AU915 in my config.h:

`// regional choices: EU868, US915, AU915, AS923, AS923_2, AS923_3, AS923_4, IN865, KR920, CN500'
'const LoRaWANBand_t Region = AU915;'
'const uint8_t subBand = 0;  // For US915, change this to 2, otherwise leave on 0`

Below is a comparison of a log between two startups, no code changes or flashes, just the difference after pressing the reset button. You can see the successful activation on the left using the correct channels for AU915, but incorrect channels for AU915 on the right (https://www.thethingsnetwork.org/docs/lorawan/frequency-plans/#au915-928)

![image](https://github.com/user-attachments/assets/987086f7-1c1d-4b57-b02b-85f251ce9602)

Does the AU915 band need modifying? Or is there a switch somewhere I'm missing to restrict it to the LoRaWAN spec?

Regards,
Randy

**Comment by StevenCellist** [comment, marked as answer]
The comment in `config.h` should be slightly modified (although one could argue that you might've been able to fill it in yourself 😉):
```
const uint8_t subBand = 0;  // For US915, change this to 2, otherwise leave on 0
```
AU915 consists of multiple subbands, just like US915. If you leave it on 0, RadioLib follows the spec by selecting a random channel from each subband to see if it can join on any of the subbands. But you are definitely better off selecting the configured subband from your gateway & device console (which is subband 2, judging from the JoinAccept).

**Comment by randy197** [comment]
Legend! For some reason I wasn't thinking of the different subbands as, well.... subbands. Here I was editing the Tx and Rx frequencies haha. 

Thanks Steven!

**Comment by randy197** [comment]
Also, turns out I just needed to RTFM! (I did read this really early on, but never made the link back)

https://github.com/jgromes/RadioLib/blob/master/examples/LoRaWAN/LoRaWAN_Starter/notes.md#region

**Reply by StevenCellist** [reply]
Good to hear you got it working :) and hopefully lesson learned! (And I'll update the description there soon-ishly.)
